src/gui/shell.cpp

```diff
@@ -725,9 +725,23 @@ void Shell::handleNeovimNotification(const QByteArray &name, const QVariantList&
 			m_mouseHide = variant_not_zero(args.at(1));
 			int val = m_mouseHide ? 1 : 0;
 			m_nvim->api0()->vim_set_var("GuiMousehide", val);
-		} else if (guiEvName == "Close" && args.size() == 1) {
+		} else if (guiEvName == "Close" && (args.size() == 1 || args.size() == 2)) {
```

Comment:
This has grown in size enough that it should probably be its own function. Keeping the size of these large switch statements down helps read-ability :)

You can take a look at something like `void Shell::handleGridLine(const QVariantList& opargs)` as an example.

Ex)
```C++
void Shell::handleGuiCloseEvent(const QVariantList& args) noexcept
```

Reply:
OK, It seems better for me too.